micro_ros_diagnostic_msgs/msg/MicroROSDiagnosticStatus.msg

```diff
@@ -1,25 +1,36 @@
 # This message holds the status of an individual component of the robot.
 #
-# This message specification is a copy of
+# This message specification is inspired by
 # common_interfaces/diagnostic_msgs/msg/DiagnosticStatus.msg
 # with the difference that it comprises just one key and value
-# instead of an array of KeyValue messages.
+# instead of an array of KeyValue messages as well as avoiding
+# any strings.
 
 # Possible levels of operations.
 byte OK=0
 byte WARN=1
 byte ERROR=2
 byte STALE=3
 
+# Possible value types
+byte VALUE_BOOL=1
+byte VALUE_INT=2
+byte VALUE_DOUBLE=3
+byte VALUE_LOOKUP=10
+
 # Level of operation enumerated above.
 byte level
+
 # A description of the test/component reporting.
-string name
-# A description of the status.
-string message
+uint16 updater_id
 # A hardware unique string.
-string hardware_id
-# What to label this value when viewing.
-string key
-# A value to track over time.
-string value
+uint16 hardware_id
+
+# What key, value is transmitted.
+uint16 key
+
+byte value_type
+bool bool_value
+int64 int_value
+float64 double_value
```

Comment:
Using float (4 bytes) is usually better and safer than double (8 bytes). First, in most chips (32 bits), the read/write is atomic. Second, due to precision and how they work, most modern chips will be able to optimize them much better, rather than working with doubles.

The same goes with int64_t, wouldn't it be better to use int32_t?